packages/gatsby-source-contentful/README.md

```diff
@@ -303,6 +295,97 @@ To get **all** the `CaseStudy` nodes with ShortText fields `id`, `slug`, `title`
 
 When querying images you can use the `fixed`, `fluid` or `resize` nodes to get different sizes for the image (for example for using [gatsby-image](https://www.gatsbyjs.org/packages/gatsby-image/)). Their usage is documented at the [gatsby-plugin-sharp](https://www.gatsbyjs.org/packages/gatsby-plugin-sharp/) package. The only difference is that gatsby-source-contentful also allows setting only the `width` parameter for these node types, the height will then automatically be calculated according to the aspect ratio.
 
+## Notes on Contentful Content Models
+
+### References
+
+Contentful reference fields will create [Union type](https://graphql.org/learn/schema/#union-types) fields. To access data [Inline fragments](https://graphql.org/learn/queries/#inline-fragments) need to be used.
+
+For example, if you have `Product` and `Category` Content Models in Contentful space and `Product` has `categories` field that reference `Category` entries, you can get list of categories for product by querying:
```

Comment:
I think showing the demo query and then explaining the requirements makes it easier to understand this?

```suggestion
See below for a references example where you query for categories on a product:
```